### ISSUE TITLE
Not working on Android 14

### ISSUE THREAD

**Issue by Anjum9694** [leohearts/AlternativeUnlockXposed#9]
Not working on Android 14, says wrong pin when fake pin is entered.
Phone: Pixel 8a
Lsposed: 1.10.1 (7115)
Magisk: 28.0 (28000)

**Comment by leohearts** [comment]
I'm using Android 14 too . (PixelOS) can you attach some logcat after "Restart systemUI" and enter the fake pin for a time ?Known issue: if your lock screen doesn't have an enter button and automatically unlocks when the right length of pin entered, you may need to set the fake and real pin as the same length.On Nov 11, 2024 1:26 AM, Anjum9694 ***@***.***> wrote:
Not working on Android 14, says wrong pin when fake pin is entered.
Phone: Pixel 8a
Lsposed: 1.10.1 (7115)
Magisk: 28.0 (28000)

**Comment by Anjum9694** [comment]
https://pastebin.com/v4MSQGZq

Logcat level debug, sequence of actions logcat started, locked screen, entered fake pin, wrong pin error then real pin
(test pins in screenshot) 

![Screenshot_20241111-081929.png](https://github.com/user-attachments/assets/e7b5d9a5-ee9c-40f9-8919-b863f83c34ee)

![Screenshot_20241111-081900.png](https://github.com/user-attachments/assets/69110b8e-4ac1-4daa-96b7-30dde10d0cd9)

**Comment by leohearts** [comment]
Thanks!I've reviewed the log but nothing about lsposed or this module got logged. Please make sure module is enabled for SystemUI , "disable verbose log" in lsposed settings is unchecked, and please click "restart systemUI" in alternative unlock app, after you started making the logcat .On Nov 11, 2024 11:21 AM, Anjum9694 ***@***.***> wrote:
https://pastebin.com/v4MSQGZq
Logcat level debug, sequence of actions logcat started, locked screen, entered fake pin, wrong pin error then real pin
(test pins in screenshot)
Screenshot_20241111-081929.png (view on web)
Screenshot_20241111-081900.png (view on web)

**Comment by leohearts** [comment]
![Screenshot_20241111-124529.png](https://github.com/user-attachments/assets/19a551a5-fbd7-4996-8a4f-90b134aef6d6)

![Screenshot_20241111-124524.png](https://github.com/user-attachments/assets/fedd7b97-2226-409d-98e8-9bd6191aa643)

**Comment by leohearts** [comment]
As Xposed code runs in the target app scope, per app log can't record the actual behavior in SystemUI. Anyway, did you notice any crash or something?On Nov 11, 2024 3:38 PM, Anjum9694 ***@***.***> wrote:
Steps taken:
I installed the mod version of lsposed 7224, logcat per app alternativeunlock
https://pastebin.com/MaE0vPKD

**Comment by Anjum9694** [comment]
[Log_2024-11-11_14-08-12.txt](https://github.com/user-attachments/files/17700006/Log_2024-11-11_14-08-12.txt)

There you go.. it doesn't crash or anything it just does nothing

**Comment by leohearts** [comment]
please include debug level logs. the right way to make a logcat is using adb logcat > logcat.txt, or , if you prefer any tools to do it , make sure the log level is debug. 
looks like the hook is actually working, so it shouldn't go wrong. but only ' I ' level log is captured so we can't dig further. 
if it's some permission issue, version 1.1v2 https://github.com/leohearts/AlternativeUnlockXposed/releases/tag/1.1v2 may helps.

**Comment by leohearts** [comment]
got the problem. fix is on the way :3

**Comment by leohearts** [comment]
```log
avc:  denied  { search } for  name="tmp"
```
we got the issue. fix on the way :3

**Comment by leohearts** [comment]
https://github.com/leohearts/AlternativeUnlockXposed/releases/tag/1.4

**Comment by Anjum9694** [comment]
[Log_2024-11-11_20-33-49.txt](https://github.com/user-attachments/files/17704457/Log_2024-11-11_20-33-49.txt)
Installed latest version now it doesn't seem to save any pins so I can't test it..

**Comment by Anjum9694** [comment]
> please include debug level logs. the right way to make a logcat is using adb logcat > logcat.txt, or , if you prefer any tools to do it , make sure the log level is debug. looks like the hook is actually working, so it shouldn't go wrong. but only ' I ' level log is captured so we can't dig further. if it's some permission issue, version 1.1v2 https://github.com/leohearts/AlternativeUnlockXposed/releases/tag/1.1v2 may helps.

Since you've mentioned it's a permission issue, I went back to this version and set my SELINUX to permissive and now it seems to work! The fake PIN worked fine!

**Comment by leohearts** [comment]
> > please include debug level logs. the right way to make a logcat is using adb logcat > logcat.txt, or , if you prefer any tools to do it , make sure the log level is debug. looks like the hook is actually working, so it shouldn't go wrong. but only ' I ' level log is captured so we can't dig further. if it's some permission issue, version 1.1v2 https://github.com/leohearts/AlternativeUnlockXposed/releases/tag/1.1v2 may helps.
> 
> Since you've mentioned it's a permission issue, I went back to this version and set my SELINUX to permissive and now it seems to work! The fake PIN worked fine!

In fact I did thought to let you try setting it to permissive for testing only, but it's pretty insecure to do so.
seems like it's still some selinux issue :
```
11-11 20:34:15.828  1447  1447 W android.bg: type=1400 audit(0.0:52277): avc:  denied  { read } for  name="e06a0f" dev="dm-57" ino=163878 scontext=u:r:system_server:s0 tcontext=u:object_r:system_data_root_file:s0 tclass=file permissive=0
```
I'll try to dig into it later :( btw, which root method are you using?

**Comment by Anjum9694** [comment]
Yeah it was temporary. After setting to permissive I opened the app to save the config then restarted system UI, then set it back to enforced. The fake pin still works. So I guess it somewhat works so thanks anyway.
